Feature: Implement tests for category, products

CeNTePeDe opened the pull request.

CeNTePeDe committed:
added
 - success message
 - HTTPException
 - added id to url

added mock for db
test parse function

created test folder:
- conftest.py
- test_api_category
- test_api_products

slavadan commented on conftest.py:
```diff
@@ -0,0 +1,23 @@
+@pytest.fixture()
+def client():
+    with TestClient(app) as c:
+        yield c
```
не as c

slavadan commented on conftest.py:
```diff
@@ -0,0 +1,23 @@
+    def fake_db():
+        return db
+
+    monkeypatch.setattr("core.constant_variables.db", fake_db)
```
test_database

slavadan commented on database/connection_categories.py:
```diff
@@ -13,13 +13,14 @@
 
-    def __init__(self):
-        self.collection = db["categories"]
+    def __init__(self, collection):
```
тайпхинт

slavadan commented on database/connection_categories.py:
```diff
@@ -32,21 +33,22 @@ def create_item(self, category: CategoryModel) -> CategoryModel:
+        category_model = self.collection.find_one({"_id": category.inserted_id})
+        return CategoryModel(**category_model)
 
     def get_all_item(self) -> list[CategoryModel]:
```
когда items ты же возвращаешь не 1 элемент

slavadan commented on database/connection_products.py:
```diff
@@ -21,33 +21,35 @@ def get_all_item(self) -> list[ProductModel]:
         collection = self.collection.find()
```
items

slavadan commented on tests/factories.py:
```diff
@@ -0,0 +1,24 @@
+
+    name_product = "test_product"
+    picture_link = factory.Faker("url")
+    price = "345.9"
```
цену рандомную делай

slavadan commented on tests/test_api_category.py:
```diff
@@ -0,0 +1,49 @@
+
+logger = logging.getLogger(__name__)
+
+URL = "http://localhost:8000/api/categories/"
```
в константы или вообще в настройки

slavadan commented on tests/test_api_category.py:
```diff
@@ -0,0 +1,49 @@
+def test_delete_category(client):
+    category = "new_category"
+    response = client.delete(url=URL + f"{category}")
+    assert response.status_code == 204
```
сделай раздление по AAA.

slavadan commented on tests/test_api_product.py:
```diff
@@ -0,0 +1,69 @@
+from tests.factories import CategoryFactory, ProductFactory
+
+logger = logging.getLogger(__name__)
+URL = "http://localhost:8000/api/products/"
```
в константы или вообще еще лучше в настройки

slavadan commented on tests/test_parse_product.py:
```diff
@@ -0,0 +1,18 @@
+from parsers.parse_product import gather_data
+
+URL = "http://localhost:8000/api/products/parser"
+url_for_parser = "https://www.lamoda.by/c/4170/clothes-maternityclothes/"
```
в константы или лучше вообще в тесты

CeNTePeDe committed:
added
- typehint
- exception to category_routers.py
- exception to product_routers.py
- url to constant variables
- random price to ProductFactory
